### ISSUE TITLE
AI will hunt down aircraft with units that can't target them

### ISSUE THREAD

**Issue by Mailaender** [OpenRA/OpenRA#3468]
##

**Comment by pchote** [comment]
Is this still an issue?  I don't recall ever seeing this happen in the last year or so.

**Comment by abcdefg30** [comment]
I saw a light tank hunting some yaks, but on release-20150919.

On the other hand, some time after the light tank rocket infantry joined, so I guess the AI just sent a whole squad with a few members that can shoot at air targets.
Actually wouldn't those extra units (that can't attack air) count as guards for the anti-air guys?

**Comment by GraionDilach** [comment]
It could also be that the light tank decided to hunt the yak while that was on an airfield but since it flew off and the tank simply did not drop the target.

**Comment by abcdefg30** [comment]
I don't think that was the case in this situation, the yak just came back from an attack and was circling above an airfield. But I guess the situation described (by you) can happen, too.

**Comment by pchote** [comment]
Is this still a problem now that we drop targets under the shroud?

**Comment by Mailaender** [comment]
Yes, see this exagerrated situation:

![image](https://cloud.githubusercontent.com/assets/756669/12379930/2b320386-bd65-11e5-8e2d-02815b0b9368.png)

The AI harvester (as it has a gun) is trying to hunt my helicopter, because it attacked the power plant, but the harvester MG can't even target the helicopter while it is airborne. This probably has something to do with the actor upgrades which control the targeting nowadays.

**Comment by reaperrr** [comment]
@pchote is this already fixed by your AutoTarget refactor or will this need further changes?

**Comment by reaperrr** [comment]
My AI squad changes might have fixed this (they look for *enabled* target types now). Can anyone confirm (or deny)?

**Comment by GraionDilach** [comment]
I can't confirm this being fixed. AI still stalls if the attacking squad is confronted by an aircraft or an undetected stealth unit.

**Comment by matjaeck** [comment]
> AI will hunt down aircraft with units that can't target them

This is fixed in RA. I just tested it and only rockets soldiers tried to hunt down my yaks while the other units that can't target air units didn't move.

> AI still stalls if the attacking squad is confronted by an aircraft or an undetected stealth unit.

If there is such an issue, it should be handled in its own ticket I think.

**Comment by GraionDilach** [comment]
>I just tested it and only rockets soldiers tried to hunt down my yaks while the other units that can't target air units didn't move.

This is exactly the bug the ticket's about!

**Comment by matjaeck** [comment]
@GraionDilach This ticket is about 

> AI will hunt down aircraft with units that can't target them 

and I can't reproduce this. If you can, please ask to reopen this.

**Comment by GraionDilach** [comment]
The AI units still keep trying to order all units to hunt down air units but only those respond now which are able to.

You have reproduced this. You said it so. It's your problem that you're unable to connect the dots.

**Comment by matjaeck** [comment]
No, I said I can not reproduce the issue that AI hunts down aircraft with units that can not target them. I never said a word about target lines, neither did the issue. Your attempts to ridicule me won't help with the problem you seem to have with the decision to close the issue. I can not help you, and it looks to me that you have no interest in a productive conversation so I'll quit talking with you now.

**Comment by matjaeck** [comment]
People report on discord that this is stilll an issue in latest release.
